src/htm/algorithms/SpatialPooler.cpp

```diff
@@ -866,24 +866,40 @@ void SpatialPooler::inhibitColumnsLocal_(const vector<Real> &overlaps,
 
 
       if (wrapAround_) {
+        numNeighbors = 0;  // In wrapAround, number of neighbors to be considered is solely a function of the inhibition radius, the number of dimensions, and of the size of each of those dimenions
+        UInt predN = 1;
+        for (UInt i = 0; i<columnDimensions_.size();i++) {
+          UInt diam = 2*inhibitionRadius_ + 1;
+          predN *= ( diam + ((columnDimensions_[i] - diam) * ((UInt) (columnDimensions_[i] - diam) >> (shftInt))) ); // max(diam, columDimenions_[i]);
+        }
+        predN -= 1;
+        numNeighbors = predN;
+        const UInt numActive_wrap = (UInt)(0.5f + (density * (numNeighbors + 1)));
+
         for(auto neighbor: WrappingNeighborhood(column, inhibitionRadius_,columnDimensions_)) { //TODO if we don't change inh radius (changes only every isUpdateRound()),
 		// then these values can be cached -> faster local inh
-          if (neighbor == column) {
-            continue;
-          }
-          numNeighbors++;
 
-          const Real difference = overlaps[neighbor] - overlaps[column];
-          if (difference > 0 || (difference == 0 && activeColumnsDense[neighbor])) {
-            numBigger++;
+	  UInt a = neighbor; // don't abuse the "operator*" function
+	  if (a == column) {
```

Comment:
ditto, is this micro-opt worth the trouble?

Reply:
> ditto, is this micro-opt worth the trouble?

With regards specifically to "a = neighbor", I'll say this:

I had never done C++ before this, and I'm unfamiliar with how operators work there, but if the "operator*()" function is called each time "neighbor" is used (that is, if the "operator*()" must be called to return a "UInt" each time "neighbor" is referenced), then I'd say that yes, it is worth the trouble. The linux "perf" tool  (http://www.brendangregg.com/perf.html) reported that a lot of time was being spent on the "operator*()" function in the original code, and this amount of time was significantly reduced after the code changes in the pull-request were made. I don't specifically remember if the "a = neighbor" change contributed much to the decrease, but I can benchmark it and get back to you.

With regards to other "a = vector[number]" optimizations, I'll say this:

I did most of my benchmarking with the DEBUG flag turned on, and, at least in that scenario: (a) each vector[number] reference in the code resulted in a lot of assembler instructions; and (b) the "a = vector[number]" changes resulted in noticeable speed improvements. I'll benchmark everything with the DEBUG flag turned off and let you know. The only reason I hadn't done it before is because compiling is slow in my machine (and so is running the hotgym benchmark) and that makes it a pain to benchmark each code change separately.

Reply:
> he linux "perf" tool[..]reported that a lot of time was being spent on the "operator*()" function in the original code, and this amount of time was significantly reduced after the code changes in the pull-request were made

I love the level of details you went to, really appreciate that. 

> I did most of my benchmarking with the DEBUG flag turned on, 

I think this could be the culpit, the compiler is quite apt at the low-level changes (esp at the `-O3, -flto`, so the code runs much faster at that '-DCMAKE_BUILD_TYPE=Release' level. 

> I'll benchmark everything with the DEBUG flag turned off and let you know. The only reason I hadn't done it before is because compiling is slow  

yes, I know, sorry. But we'll have to figure impact of each (type of) change.